### PR TITLE
Create notebook introducing workspace resources

### PR DESCRIPTION
**Pull request by emmarogge**
Create notebook introducing the types of referenced and controlled resources which can be added to Verily Workbench resources. Provide widgets for adding each type of referenced and controlled resource to the user's current workspace.

### CODE REVIEW

File: first_hour_on_terra/workspace_resource_examples.ipynb [hunk 1]
```diff
@@ -0,0 +1,1403 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "#### 1. Add a GitHub repository\n",
+    "\n",
+    "Both public and private GitHub repositories can be added to data collections. In order to add a private GitHub repository, you must first [set up your Terra SSH key](https://terra-docs.api.verily.com/docs/how_to_guides/terra_ssh_key_guide) and should provide a repo URL of the format `git@github.com` instead of `http://`. No additional setup is required to add a public GitHub repository.\n",
```
Comment (wnojopra) [inline first_hour_on_terra/workspace_resource_examples.ipynb]:
```suggestion
    "Both public and private GitHub repositories can be added to data collections. In order to add a private GitHub repository, you must first [set up your Terra SSH key](https://terra-docs.api.verily.com/docs/how_to_guides/terra_ssh_key_guide) and should provide a repo URL of the format `git@github.com` and not `http://`. No additional setup is required to add a public GitHub repository.\n",
```

File: first_hour_on_terra/workspace_resource_examples.ipynb [hunk 2]
```diff
@@ -0,0 +1,1403 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "class AddGcsBucketWidget(object):\n",
+    "    def __init__(self):\n",
+    "        self.label = widgets.Label(value = 'Please provide appropriate values in the input boxes.')\n",
```
Comment (wnojopra) [inline first_hour_on_terra/workspace_resource_examples.ipynb]: Just like with the data collections notebook, I think it needs to be more clear what the user is expected to input here. Please add instructions here that tells the user that they're expected to make up a repo name and description and give examples of good names. Also give an example of a github url (maybe a public one). I think the cloning options also need some explanations.

File: first_hour_on_terra/workspace_resource_examples.ipynb [hunk 3]
```diff
@@ -0,0 +1,1403 @@
...
+    "\n",
+    "create_gcs_bucket_widget = CreateGcsBucketWidget()\n",
+    "display(create_gcs_bucket_widget.vb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 3. Create a GCP notebook\n",
```
Comment (wnojopra) [inline first_hour_on_terra/workspace_resource_examples.ipynb:1255]: Even though the command is called `create gcp-notebook`, I really think we should call it a cloud environment. https://terra-docs.api.verily.com/docs/how_to_guides/cloud_environments/#accessing-the-terra-command-line-tool-from-your-cloud-environment for example correctly calls it a cloud environment even though the command line example calls it a gcp-notebook.

In fact, I'm going to file a ticket to rename it in the CLI.

**Review by wnojopra — CHANGES_REQUESTED** [review]

File: first_hour_on_terra/workspace_resource_examples.ipynb (hunk 1 above)
Comment (emmarogge) [reply first_hour_on_terra/workspace_resource_examples.ipynb]: done!

File: first_hour_on_terra/workspace_resource_examples.ipynb (hunk 3 above)
Comment (emmarogge) [reply first_hour_on_terra/workspace_resource_examples.ipynb:1255]: awesome! I will update this accordingly once the CLI change lands.

File: first_hour_on_terra/workspace_resource_examples.ipynb (hunk 2 above)
Comment (emmarogge) [reply first_hour_on_terra/workspace_resource_examples.ipynb]: I have now gone in and provided detailed explanations for each of the input parameters for each widget. It makes for a longer notebook, overall, but hopefully one that's more useful/less frustrating for users.

**Comment by emmarogge** [comment]
Closing this PR, continuing in #33

**Pull request closed without merging**